feat(api): add renewable clusters to API endpoints

hdinia opened the pull request:
fix #1792

hdinia committed:
feat(api): add renewables clusters api endpoints

ghost commented:
### Features

*  **api:** add renewables clusters api endpoints ([f9e9cfa](https://github.com/AntaresSimulatorTeam/AntaREST/commit/f9e9cfa2ef442aee1575afde7ce3fb664f558aa1))
*  **api:** add backport API endpoints for renewable cluster management ([88a7cae](https://github.com/AntaresSimulatorTeam/AntaREST/commit/88a7caeb483f49e40c4adfbf8e7b82f2dac1a8bf))
*  **variant:** avoid removing binding constraints when a renewable cluster is removed ([b9baeb1](https://github.com/AntaresSimulatorTeam/AntaREST/commit/b9baeb192ff996b07a66ea92c8fe2d92938c8860))


### Chore

* correct comment in unit test ([4281adb](https://github.com/AntaresSimulatorTeam/AntaREST/commit/4281adb7c2678a7a8aee6e50126da7d0d0c1b888))


### Bug Fixes

*  **command:** ensure the area exist before apply renewable cluster deletion ([f22517b](https://github.com/AntaresSimulatorTeam/AntaREST/commit/f22517b9b6ad870e8b52f127ffa40176d28e0391))
*  **pydantic:** fix wrong pydantic models ([274a165](https://github.com/AntaresSimulatorTeam/AntaREST/commit/274a16538cc761d4b4e2d1e2a1670620d3ef0477))


### Tests

*  **renewable:** correct integration test of renewable clusters ([fbbf7ce](https://github.com/AntaresSimulatorTeam/AntaREST/commit/fbbf7ce35c4c05b3f1e4d415620b461ac86a00bc))
*  **variants:** correct integration test of renewable clusters ([5198126](https://github.com/AntaresSimulatorTeam/AntaREST/commit/5198126a9234254e3760f522901fb6d00d0d8fe8))


### Styles

*  **renewable:** correct code style ([8a8fd76](https://github.com/AntaresSimulatorTeam/AntaREST/commit/8a8fd76bbb9051737f9cf6cbd3b2c389afcecad2))


### Documentation

*  **variant:** add a comment in `CreateRenewablesCluster` class about the `renewable-generation-modelling` checking ([8a2ac6f](https://github.com/AntaresSimulatorTeam/AntaREST/commit/8a2ac6f5a55825542f169c02eb5707470a0b32de))


### Contributors

<a href="https://github.com/hdinia">hdinia</a>, <a href="https://github.com/laurent-laporte-pro">laurent-laporte-pro</a>, <a href="https://github.com/MartinBelthle">MartinBelthle</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

MartinBelthle commented:
I'll wait for you to fix the lint and the tests so i can start the review :)

laurent-laporte-pro committed:
chore: correct comment in unit test

laurent-laporte-pro committed:
fix(command): ensure the area exist before apply renewable cluster deletion

laurent-laporte-pro committed:
test(renewable): correct integration test of renewable clusters

laurent-laporte-pro committed:
style(renewable): correct code style

laurent-laporte-pro committed:
test(variants): correct integration test of renewable clusters

laurent-laporte-pro committed:
feat(api): add backport API endpoints for renewable cluster management

those endpoints are indicated as deprecated in the documentation.

laurent-laporte-pro committed:
docs(variant): add a comment in `CreateRenewablesCluster` class about the `renewable-generation-modelling` checking

laurent-laporte-pro committed:
feat(variant): avoid removing binding constraints when a renewable cluster is removed

There is no binding constraint with renewable clusters, only with thermal clusters.

MartinBelthle committed:
fix(pydantic): fix wrong pydantic models
